app/assets/javascripts/lib/ranger_connection.js.coffee

```diff
@@ -3,19 +3,21 @@ query = {stream: stream}
 address = "#{gon.ranger_host}:#{gon.ranger_port}?" + $.param(query, true)
 barongPath = "#{gon.barong_domain}" + "/api/v1/sessions/jwt"
 window.ranger = new RangerWebSocket address
-$.ajax({
-  type: "POST",
-  url: barongPath,
-  async: true,
-  xhrFields: {
-    withCredentials: true
-  },
-  success: (res) ->
-    ranger.bind 'open', ->
-      console.info 'Connection to Ranger has been established'
-      return if not gon.user
-      console.info 'Solving authorization challenge by sending JWT token'
-      ranger.send JSON.stringify {jwt: "Bearer #{res}"}
-    ranger.bind 'error', ->
-      alert("Sorry, couldn't establish connection with WebSocket API. If problem persists, please contact support.")
-})
+ranger.bind 'open', ->
+  getCookie = (name) ->
+    nameEQ = name + "="
+    ca = document.cookie.split(";")
+    i = 0
+
+    while i < ca.length
+      c = ca[i]
+      c = c.substring(1, c.length)  while c.charAt(0) is " "
+      return c.substring(nameEQ.length, c.length)  if c.indexOf(nameEQ) is 0
+      i++
+    null
```

Comment:
There is case when this function returns null, it means that we need to check that cookie is not null before sending it to ranger. Why would we try to solve auth challenge if we don't even have jwt token cookie?